linux/utils/install_uninstall_package.yml

```diff
@@ -130,34 +134,24 @@
             allow_downgrade: true
             state: "{{ package_state }}"
           delegate_to: "{{ vm_guest_ip }}"
-          when: guest_os_ansible_pkg_mgr | lower == "yum"
+          when: guest_os_ansible_pkg_mgr == "yum"
 
+        # Ansible has dnf and dnf5 modules based on different dnf version. Fedora 41 has bumped to dnf5, but Ansible dnf5 module
+        # could not work well on Fedora 41. So here replaces dnf modules with dnf commands to manage packages
         - name: "{{ package_manage_op }} packages on {{ guest_os_ansible_distribution }} {{ guest_os_ansible_distribution_ver }}"
-          when: guest_os_ansible_pkg_mgr | lower == "dnf"
+          when: guest_os_ansible_pkg_mgr in ['dnf', 'dnf5']
           block:
-            - name: "Get DNF version on {{ guest_os_ansible_distribution }} {{ guest_os_ansible_distribution_ver }}"
-              ansible.builtin.shell: "dnf --version | head -n 1 | cut -d '.' -f 1"
-              register: dnf_major_version_result
-              ignore_errors: true
-              delegate_to: "{{ vm_guest_ip }}"
-
-            - name: "Set fact of DNF major version"
+            - name: "Set facts of packages and command for {{ guest_os_ansible_distribution }}"
               ansible.builtin.set_fact:
-                dnf_major_version: "{{ dnf_major_version_result.stdout }}"
-              when:
-                - dnf_major_version_result.rc is defined
-                - dnf_major_version_result.rc | int == 0
-                - dnf_major_version_result.stdout is defined
-                - dnf_major_version_result.stdout
+                package_manage_cmd: |-
+                  {%- if package_state in [ 'installed', 'present', 'latest' ] -%}
+                    dnf install -y {{ package_manage_list_str }}
```

Comment:
Should it use {{ guest_os_ansible_pkg_mgr }} instead of directly setting it to "dnf"?

Reply:
No, it can't. guest_os_ansible_pkg_mgr could be dnf or dnf5, but the command should be dnf.